### PR TITLE
WiP: galp5 build based on librem_11 (GOP based iGPU)

### PR DESCRIPTION
**Pull request by tlaurion**
@loftlifter31 this is my take helping you get the base to make this forward.

Superseeds https://github.com/linuxboot/heads/pull/1766

### CODE REVIEW

**Comment by tlaurion** [comment]
As per DCO CI warning https://github.com/linuxboot/heads/pull/1774/checks?check_run_id=29617868402, resigning all comits, making us co-authors.

Please sign commits form local git repo. Steal reuse my commit content and force push to your branch under #1766  (currently your master branch, otherwise ropen another PR, closing #1766 as per your choosing)

**Comment by tlaurion** [comment]
Squashing all your commits to original work (no author etc)

**Comment by tlaurion** [comment]
Rebase+squash+signoff detailed under https://github.com/linuxboot/heads/pull/1774/commits/4418282b366f11bac781d9f0000b8dbc0e49aa6a commit message

File: blobs/galp5/ifd.bin
Comment (tlaurion) [inline blobs/galp5/ifd.bin:1]: Where is this coming from @loftlifter31 ?

File: blobs/galp5/me.bin
Comment (tlaurion) [inline blobs/galp5/me.bin:1]: Where is this coming from @loftlifter31 ?

File: modules/coreboot [hunk 1]
```diff
@@ -97,6 +97,13 @@ coreboot-dasharo_commit_hash := 3a9aa3a4692f3dd49732f5b4e3ec54be385f0969
 coreboot-dasharo_patch_version := unreleased
 $(eval $(call coreboot_module,dasharo,))
 
+# System76 maintain their own fork
+coreboot-system76_repo := https://github.com/system76/coreboot
+coreboot-system76_commit_hash := 6da534f92044c684139675a599eef92d077b2b36
```
Comment (tlaurion) [inline modules/coreboot:102]: This is tip of fork at time of writing https://github.com/system76/coreboot/commit/6da534f92044c684139675a599eef92d077b2b36

File: config/coreboot-galp5.config
Comment (tlaurion) [inline config/coreboot-galp5.config:1]: This was generated using modules/coreboot Makefile helper. Feel free to finetune with:
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:latest -- make BOARD=qemu-coreboot-whiptail-tpm2-hotp coreboot.modify_and_save_oldconfig_in_place

File: config/linux-galp5.config
Comment (tlaurion) [inline config/linux-galp5.config:1]: This was generated using modules/linux Makefile helper. Feel free to finetune with:
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:latest -- make BOARD=galp5 linux.modify_and_save_oldconfig_in_place

File: config/linux-galp5.config [hunk 2]
```diff
@@ -0,0 +1,3141 @@
...
+#
+# CONFIG_FB_CIRRUS is not set
+# CONFIG_FB_PM2 is not set
+# CONFIG_FB_CYBER2000 is not set
+# CONFIG_FB_ARC is not set
+# CONFIG_FB_ASILIANT is not set
+# CONFIG_FB_IMSTT is not set
+# CONFIG_FB_VGA16 is not set
+# CONFIG_FB_VESA is not set
+CONFIG_FB_EFI=y
```
Comment (tlaurion) [inline config/linux-galp5.config:1760]: Absolutely needed

File: config/coreboot-galp5.config [hunk 3]
```diff
@@ -0,0 +1,880 @@
...
+
+#
+# Devices
+#
+CONFIG_HAVE_LINEAR_FRAMEBUFFER=y
+CONFIG_HAVE_FSP_GOP=y
+# CONFIG_VGA_ROM_RUN is not set
+CONFIG_RUN_FSP_GOP=y
+# CONFIG_NO_GFX_INIT is not set
+CONFIG_NO_EARLY_GFX_INIT=y
```
Comment (tlaurion) [inline config/coreboot-galp5.config:554]: Should work, as for librem_11

**Review by tlaurion — COMMENTED** [review]
Here you go, commented in code @loftlifter31

File: boards/galp5/galp5.config [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+CONFIG_CAIRO=y
+CONFIG_FBWHIPTAIL=y
+CONFIG_HOTPKEY=y
+
+CONFIG_LINUX_USB=y
+CONFIG_MOBILE_TETHERING=y
+
+export CONFIG_TPM=y
+export CONFIG_SUPPORT_USB_KEYBOARD=y
```
Comment (tlaurion) [inline boards/galp5/galp5.config:30]: Not sure how the keyboard is connected. If PS2, remove.

File: config/coreboot-galp5.config [hunk 5]
```diff
@@ -0,0 +1,880 @@
...
+CONFIG_DCACHE_RAM_BASE=0xfef00000
+CONFIG_DCACHE_RAM_SIZE=0x80000
+CONFIG_C_ENV_BOOTBLOCK_SIZE=0x40000
+CONFIG_DCACHE_BSP_STACK_SIZE=0x40400
+CONFIG_MAX_ACPI_TABLE_SIZE_KB=144
+CONFIG_HAVE_INTEL_FIRMWARE=y
+CONFIG_MRC_SETTINGS_CACHE_SIZE=0x10000
+CONFIG_DRIVERS_INTEL_WIFI=y
+CONFIG_IFD_BIN_PATH="@BLOB_DIR@/galp5/ifd.bin"
+CONFIG_ME_BIN_PATH="@BLOB_DIR@/galp5/me.bin"
```
Comment (tlaurion) [inline config/coreboot-galp5.config:154]: Points to your blobs. Not checked validity

File: config/coreboot-galp5.config [hunk 6]
```diff
@@ -0,0 +1,880 @@
...
+CONFIG_ACPI_NO_CUSTOM_MADT=y
+CONFIG_ACPI_COMMON_MADT_LAPIC=y
+CONFIG_ACPI_COMMON_MADT_IOAPIC=y
+CONFIG_HAVE_ACPI_TABLES=y
+CONFIG_ACPI_LPIT=y
+CONFIG_BOOT_DEVICE_SPI_FLASH=y
+CONFIG_BOOT_DEVICE_MEMORY_MAPPED=y
+CONFIG_BOOT_DEVICE_SUPPORTS_WRITES=y
+CONFIG_RTC=y
+CONFIG_HEAP_SIZE=0x100000
```
Comment (tlaurion) [inline config/coreboot-galp5.config:735]: Might need to be bumped to 0x100000 otherwise no bootsplash

**Review by tlaurion — COMMENTED** [review]
Some more comments

File: blobs/galp5/ifd.bin
Comment (loftlifter31) [reply blobs/galp5/ifd.bin:1]: https://github.com/system76/firmware-open/tree/master/models/galp5

File: blobs/galp5/me.bin
Comment (loftlifter31) [reply blobs/galp5/me.bin:1]: https://github.com/system76/firmware-open/tree/master/models/galp5

File: boards/galp5/galp5.config (hunk 4 above)
Comment (loftlifter31) [reply boards/galp5/galp5.config:30]: It's a laptop so the keyboard is built in.

Comment (tlaurion) [reply boards/galp5/galp5.config:30]: > It's a laptop so the keyboard is built in.

Doesn't mean much unfortunately without having laptop in hands. It either needs to be ps2/usb and if usb, drivers needs to loaded early under Heads.

Hence why I raise your attention there.
Its cheaper for manufacturers nowaday to do usb, many do that. I dunno their implementation.

File: blobs/galp5/me.bin
Comment (tlaurion) [reply blobs/galp5/me.bin:1]: Ideally there is a script downloading me.rom from a specific commit, so that this repo doesn't grow with blobs, just like for other boards, see blobs scripts.

Alternatively, as other coreboot forks do, there is seperate blobs. Those might be under coreboot blobs already downloaded, I haven't verified either.

Tldr:we would merge a blog of 5mb in this tree that will change later.

See past blobs in their repo, changed 5+ times already https://github.com/system76/firmware-open/commits/master/models/galp5/me.rom

File: config/linux-galp5.config (hunk 2 above)
Comment (tlaurion) [reply config/linux-galp5.config:1760]: > I wasn't sure what to put in this config file. The system76 build process doesn't use it since they use coreboot with an edk2 payload. I tried copying the config for the running kernel but that has a ton of multimedia drivers and created a monster sized kernel 80+ Mb. Too big for the rom. So in the end I just copied the linux config from another board. x230 I think since it uses close to the same kernel version 5.15 How will I know what FB to set here?

Does it work?
It should be GOP (provided by fsp for iGPU to drive coreboot FB) and then efifb as said.

Only librem 11 is goon driven under heads today, everything else is libgfxinit driven under coreboot.

So if GOP initialize 2d FB enough so coreboot is happy with linear FB init, then efifb from kernel can reuse it, then you would be good to go on that level.


Test results?

Comment (tlaurion) [reply config/linux-galp5.config:1760]: Also this or is not based on x230 config anymore. Neither the same kernel ver, coreboot ver nor configs.

This is derivate of librem 11, and latest commit from their coreboot fork.

Have you reviewed all the changes?

**Comment by tlaurion** [comment]
> Here you go, commented in code @loftlifter31. CircleCI will produce ROM to be tested, as per https://osresearch.net/Downloading#downloading-heads-from-circleci but from this PR built artifacts.
> 
> WARNING: this is just first iteration, I do not own the hardware and cannot test. This is first iteration, where GOP/vbt/EFIFB will most probably need some additional work. If this doesn't boot, we can activate debbuging traces to be saved in SPI, and taking a backup there, increasing debug levels and extracting coreboot logs will push this forward.

@loftlifter31 did you test galp5 built rom artifact from this PR?

**Comment by tlaurion** [comment]
> > Here you go, commented in code @loftlifter31. CircleCI will produce ROM to be tested, as per https://osresearch.net/Downloading#downloading-heads-from-circleci but from this PR built artifacts.
> > 
> > WARNING: this is just first iteration, I do not own the hardware and cannot test. This is first iteration, where GOP/vbt/EFIFB will most probably need some additional work. If this doesn't boot, we can activate debbuging traces to be saved in SPI, and taking a backup there, increasing debug levels and extracting coreboot logs will push this forward.
> 
> @loftlifter31 did you test galp5 built rom artifact from this PR? 

@loftlifter31 afger following referred doc you should have landed to https://output.circle-artifacts.com/output/job/5c445ab2-7bfa-4ed5-8b35-1e3993322e0f/artifacts/0/build/x86/galp5/heads-galp5-v0.2.0-2303-g9638cdb.zip

**Comment by tlaurion** [comment]
Oh. Librem 11 is not a board config with tpm2 dependencies. So above rom might boot with FB and console output or not, but t+heads rom toolstack won't work. Board config needs to be adapted from another board that is tpm2, ie nv41.

**Comment by tlaurion** [comment]
> Oh. Librem 11 is not a board config with tpm2 dependencies. So above rom might boot with FB and console output or not, but t+heads rom toolstack won't work. Board config needs to be adapted from another board that is tpm2, ie nv41.

fixed under [2db9de9](https://github.com/linuxboot/heads/pull/1774/commits/2db9de920a5aa2ebbde4d4553bd21460869e42c6)

File: boards/galp5/galp5.config (hunk 4 above)
Comment (tlaurion) [reply boards/galp5/galp5.config]: @loftlifter31 optional, runtime configurable option under https://github.com/linuxboot/heads/pull/1774/commits/76d2dc8443e27d0f21e44ba67c85227e3cfe5793

File: boards/galp5/galp5.config [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+#Enable DEBUG output
+#export CONFIG_DEBUG_OUTPUT=y
+#export CONFIG_ENABLE_FUNCTION_TRACING_OUTPUT=y
+#Enable TPM2 pcap output under /tmp
+#export CONFIG_TPM2_CAPTURE_PCAP=y
+
+#On-demand hardware support (modules.cpio)
+CONFIG_LINUX_USB=y
+CONFIG_LINUX_E1000=y
```
Comment (tlaurion) [inline boards/galp5/galp5.config:24]: @loftlifter31 unclear what  ethernet controller is present under HCL https://tech-docs.system76.com/models/galp5/README.html you might need to adapt

**Review by tlaurion — COMMENTED** [review]
Some more changes following previous comments

File: config/coreboot-galp5.config (hunk 6 above)
Comment (tlaurion) [reply config/coreboot-galp5.config]: @loftlifter31 added under [be1f40b](https://github.com/linuxboot/heads/pull/1774/commits/be1f40b8343de0a3bf9daf0fa04c06c26181b380)

Others might be needed see commit message

File: config/linux-galp5.config (hunk 2 above)
Comment (tlaurion) [reply config/linux-galp5.config:1760]: @loftlifter31 system76 coreboot fork seems to be missing some patches from upstream. See patches/coreboot-24.02.01 for reference, added HEAP_SIZE patch needed for bootsplash to work, others might be missing

File: config/coreboot-galp5.config [hunk 8]
```diff
@@ -758,7 +757,8 @@ CONFIG_TTYS0_LCS=3
 # CONFIG_SPKMODEM is not set
 # CONFIG_CONSOLE_NE2K is not set
 CONFIG_CONSOLE_CBMEM=y
-# CONFIG_CONSOLE_SPI_FLASH is not set
+CONFIG_CONSOLE_SPI_FLASH=y
+CONFIG_CONSOLE_SPI_FLASH_BUFFER_SIZE=0x20000
```
Comment (tlaurion) [inline config/coreboot-galp5.config:761]: @loftlifter31 this is console output (equivalent of cbmem console output) saved in SPI.
0x20000 might not be enough, and more debugging activation might be needed.

But this will permit to take a falshrom/flashprog backup externally and investigate the content in case of a bricked system and post those logs to make this go faster in case EFIFB/libgfxinit/GOP not working as expected.

**Review by tlaurion — COMMENTED** [review]
@loftlifter31 logs/trace of current PR status expected to move things formward. I think this is now a good starting base to iterate upon, unless system76 coreboot fork's latest commit (unmoving for months, not sure of their base (24.02.01? didn't check) is wrong for non UEFI setups. Then if that is the case, more work will need to be done in collaboration of system76.

File: boards/galp5/galp5.config (hunk 7 above)
Comment (loftlifter31) [reply boards/galp5/galp5.config:24]: Not sure why I would need ethernet during boot up. Once control is passed to full linux kernel that will take care of loading correct drivers, yes?

File: config/coreboot-galp5.config (hunk 6 above)
Comment (tlaurion) [reply config/coreboot-galp5.config]: Are we trying to get galp5 for your sole usage or for everybody else? That will change my involvement in this effort.

Comment (loftlifter31) [reply config/coreboot-galp5.config]: My primary concern at this point is to get it working for myself according to my needs. However, I don't see any problem with testing alternate configurations which may benefit others, such as increasing the heap size.

File: boards/galp5/galp5.config (hunk 7 above)
Comment (tlaurion) [reply boards/galp5/galp5.config:24]: > Not sure why I would need ethernet during boot up. Once control is passed to full linux kernel that will take care of loading correct drivers, yes?

This driver needs to be right (e1000e, igb whatnot) to be able to use network-init-recovery script (which syncs time through ntp and get an address from dhcp prior) from recovery shell today. It's completely on demand and nothing today does the loading of the driver automatically outside of this codepath.

Heads hopefully will evolve into squashfs+overlayfs overlays so that things can be pulled from the network (wyng, docker whatever) depending of the use case pushed by the oem down to its users (dasharo DTS being one desired case, to bypass need of ipxe: we could wget kernel +initrd and kexec into it), another one being python3 overlayfs to jump into getting system state restoration as a service.

Note that the board config as it is already included android network tethering drivers also prompting the user to be used transparently through network-init-recovery today.

Tldr: on demand network access, loading drivers not on standard boot path, measures then load additional drivers, changing PCR content and preventing heads to release secrets that were sealed to unseal only on default boot path. So loading any on-demand driver will prevent efficiently any secret to be unsealed. Same applies to recovery shell access: doing on demand operations extend PCR banks which invalidates measurements .

Todo: Getting the right Ethernet driver there is needed for on-demand networking. Gotta figure out if it's igb e1000 or e1000e

**Comment by tlaurion** [comment]
From https://github.com/linuxboot/heads/issues/1765#issuecomment-2338646515
> I got it working. Just needed to update vbt.

Can you create a branch with everything tested working on your side? Like I said, HEAP size bumped to 40000 should work even with bootsplash if GOP + efifb works. Some output would be great. Will rebase this branch on master and force push.

After that, I will let this PR bitrot until picked up to be upstreamed.

File: blobs/galp5/ifd.bin
Comment (tlaurion) [reply blobs/galp5/ifd.bin:1]: @loftlifter31 added under board config as hcl ref.

File: config/coreboot-galp5.config (hunk 6 above)
Comment (tlaurion) [reply config/coreboot-galp5.config]: Put at right value under [605386](https://github.com/linuxboot/heads/pull/1774/commits/860538611fcca62e4e522883fdfd1c037c46434a) with patch under patches/ see pr files changed.

Comment (loftlifter31) [reply config/coreboot-galp5.config]: I rebuilt heads with CONFIG_HEAP_SIZE=0x100000. Still no bootsplash at this size. Above you suggested a patch may be needed to increase to 400000 however the patch you provided is for coreboot version 24.02.01. I'm using version 4.22.01 for my build.

File: boards/galp5/galp5.config (hunk 7 above)
Comment (loftlifter31) [reply boards/galp5/galp5.config:24]: Docs say it should come with Intel NIC however mine seems to come with Realtek 8111 chipset.

File: config/coreboot-galp5.config (hunk 6 above)
Comment (tlaurion) [reply config/coreboot-galp5.config]: Why would you use 24.02.01 and not system76 latest know commit?

https://github.com/coreboot/coreboot/compare/0a280ff747b44d7a2a62206ea0c25dbcbc5ce20b...6da534f92044c684139675a599eef92d077b2b36

- 24.02.01: https://github.com/coreboot/coreboot/commit/0a280ff747b44d7a2a62206ea0c25dbcbc5ce20b
- system76 fork pointed by this RP modules/coreboot fork : https://github.com/system76/coreboot/commit/6da534f92044c684139675a599eef92d077b2b36

As said, yes, the patch that was missing is under patches to be applied under clean build under this PR: https://github.com/linuxboot/heads/blob/1e840fcb4b397ea3e87f40abcb961a41965de155/patches/coreboot-system76-unreleased/0002-increase_heap_from1mb_to4mb_when_bootpslash_enabled.patch

You would need to rebuild this PR with modules/coreboot helper:
```
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:latest -- make -d BOARD=galp5 real.remove_canary_files-extract_patch_rebuild_what_changed
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:latest -- make -d BOARD=galp5
```

Otherwise the value cannot be changed in config file.

@loftlifter31 Please open a PR for what you attempt to do. This PR uses system76 coreboot branch so that whatever bug you find along the way will be possible to report to them, leaving me only playing with Heads parts, delegating responsibilities to system76 if things coreboot don't work: just like this missing patch and others missing under patches/coreboot-24.02.01 for them to pick up. You are more then invited into pinging them to do so, linking them to the PR that made those patches landing (use github blame/history for a file in the file browser https://github.com/linuxboot/heads/tree/master/patches/coreboot-24.02.01 then clicking https://github.com/linuxboot/heads/blob/master/patches/coreboot-24.02.01/0002-increase_heap_from1mb_to4mb_when_bootpslash_enabled.patch, look left corner, expend commit message -> leads to https://github.com/linuxboot/heads/commit/119cebc44f51b0d69069bf37e9817156b96df09c, clikc PR number associated under commit message - > https://github.com/linuxboot/heads/pull/1723


System76 won't offer support for an older version. If you want them to care, you have to use and refer to their head commit, which is https://github.com/system76/coreboot/commit/6da534f92044c684139675a599eef92d077b2b36 for which those 2 patches are missing (they have to cherry-pick them and adapt them):
- https://github.com/linuxboot/heads/blob/119cebc44f51b0d69069bf37e9817156b96df09c/patches/coreboot-24.02.01/0001-prevent_bootsplash_heap_allocation_failure_to_booting.patch
- https://github.com/linuxboot/heads/blob/119cebc44f51b0d69069bf37e9817156b96df09c/patches/coreboot-24.02.01/0002-increase_heap_from1mb_to4mb_when_bootpslash_enabled.patch

---

TLDR:

Do 
```
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:latest -- make -d BOARD=galp5 real.remove_canary_files-extract_patch_rebuild_what_changed
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:latest -- make -d BOARD=galp5
```

from this branch. This will apply https://github.com/linuxboot/heads/blob/119cebc44f51b0d69069bf37e9817156b96df09c/patches/coreboot-24.02.01/0002-increase_heap_from1mb_to4mb_when_bootpslash_enabled.patch which will permit you to increase heap size and you will get boosplash.

Note that system76 still need to apply those patches: please open an issue on their side.
Also note that if someone was raising an issue for an older version of Heads: I would say to them to use latest commit and would stop caring there. Expect System76 to do the same. Talk to them about their latest commit. This is why I created this PR to aid in your journay: to anyone galp5 wanting to use Heads to be able to do so once this PR is merged.

You continuing your adventure on your local fork without pushing a branch to have eyes on.... is to say the least, non-interesting and lowers my interests helping you, helping me, helping galp5 to get heads. Thanks for your understanding.

Comment (tlaurion) [reply config/coreboot-galp5.config]: Ha, nope. Sorry, coreboot forks have their own logic, and doing real.clean* doesn't remove the .patched flag.
Working on a fix and pushing.

Comment (tlaurion) [reply config/coreboot-galp5.config]: Ha. Realized the two patches were abandoned and https://review.coreboot.org/c/coreboot/+/83895 was merged instead.... @JonathonHall-Purism

File: config/linux-galp5.config (hunk 2 above)
Comment (loftlifter31) [reply config/linux-galp5.config:1760]: CONFIG_FB_EFI=y works ok in the linux.config file. If additional info is needed on this, I can post the relevant lines from the compile config file used to build the kernel that shipped with the laptop.

File: config/coreboot-galp5.config (hunk 6 above)
Comment (tlaurion) [reply config/coreboot-galp5.config]: See [127e5df](https://github.com/linuxboot/heads/pull/1774/commits/127e5df6dadcc9ecbcd3c8f29e9e755f131c38b7) and https://github.com/linuxboot/heads/pull/1795

There should be no need to increase HEAP anymore.

Comment (tlaurion) [reply config/coreboot-galp5.config]: Neglect https://github.com/linuxboot/heads/pull/1774#discussion_r1755582417 referred patches, deleted in latest ommit here and under https://github.com/linuxboot/heads/pull/1795

**Comment by tlaurion** [comment]
[37f809e](https://github.com/linuxboot/heads/pull/1774/commits/37f809e156c4cf0c8fce898c478f435bf33b964d)

Tested equivalent under https://github.com/linuxboot/heads/pull/1795 under qemu: works without needing to change HEAP_SIZE. See commit details.

File: boards/galp5/galp5.config (hunk 7 above)
Comment (tlaurion) [reply boards/galp5/galp5.config:24]: > Docs say it should come with Intel NIC however mine seems to come with Realtek 8111 chipset.

need actual loaded driver and dependencies from booted linux os lsmod

Comment (loftlifter31) [reply boards/galp5/galp5.config:24]: driver is r8169. No listed dependencies.

File: config/coreboot-galp5.config (hunk 6 above)
Comment (loftlifter31) [reply config/coreboot-galp5.config]: Still no bootsplash with heap set to 0x400000

**Comment by loftlifter31** [comment]
> Can you create a branch with everything tested working on your side? Like I said, HEAP size bumped to 40000 should work even with bootsplash if GOP + vbt+ efifb works. Some output would be great. Will rebase this branch on master and force push.

Create a branch? Where? On my local fork? At this point everything on my fork is tested, though bootsplash and NIC are not working.

**Comment by tlaurion** [comment]
> > Can you create a branch with everything tested working on your side? Like I said, HEAP size bumped to 40000 should work even with bootsplash if GOP + vbt+ efifb works. Some output would be great. Will rebase this branch on master and force push.
> 
> Create a branch? Where? On my local fork? At this point everything on my fork is tested, though bootsplash and NIC are not working.


[37f809e](https://github.com/linuxboot/heads/pull/1774/commits/37f809e156c4cf0c8fce898c478f435bf33b964d) is supposed to fix bootsplash. This PR includes fixes for your reported issues. Putting this as draft, seems we are not alligned to make this first candidate and make this work for others as all other boards.
